### ISSUE TITLE
DockerOperator loosing some initial container output

### ISSUE THREAD

**Issue by the-horhe** [apache/airflow#9190]
**Apache Airflow version**: 1.10.10

**Environment**: official docker image apache/airflow:1.10.10, docker api version: 1.40

**What happened**:
Don't see any container output in task logs when using DockerOperator with short living containers.

**What you expected to happen**:
Expected to see container stdout/stderr

It seems the problem is here: https://github.com/apache/airflow/blob/v1-10-stable/airflow/operators/docker_operator.py#L232
1. Containers is being started
2. We attach to container output

But all output between steep 1 and to 2 seems to be lost.

I found a simple solution - adding `logs=True` to self.cli.attach fixes the problem
https://github.com/apache/airflow/blob/v1-10-stable/airflow/operators/docker_operator.py#L235
According to docker API documentation (https://docs.docker.com/engine/api/v1.40.yaml - line 6052) solution seems legit.

**How to reproduce it**:

```
from datetime import timedelta
from airflow.utils.dates import days_ago
from airflow import DAG
from airflow.operators.docker_operator import DockerOperator


default_args = {
    'owner': 'airflow',
    'depends_on_past': True,
    'wait_for_downstream': True,
    'catchup': True,
    'start_date': days_ago(2),
    'retries': 1,
    'retry_delay': timedelta(minutes=5),
}
dag = DAG(
    'docker_dag',
    default_args=default_args,
    description='',
    schedule_interval=timedelta(days=1),
)

print_hello_world = DockerOperator(
    task_id='print_hello_world',
    image='centos:latest',
    api_version='auto',
    auto_remove=True,
    command='/bin/bash -c \'echo "HELLO WORLD!"\'',
    docker_url='unix://var/run/docker.sock',
    network_mode='bridge',
    dag=dag,
)
```
Output without logs=True:
![image](https://user-images.githubusercontent.com/12816025/84119285-d0704600-aa3c-11ea-8f4b-f7a12a889c24.png)

Output without logs=True but with slow starting container (command `/bin/bash -c \'sleep 5 && echo "HELLO WORLD!"\'`):
![image](https://user-images.githubusercontent.com/12816025/84119528-2ba23880-aa3d-11ea-9186-bc0a9e2d0ffe.png)
^ ok - notice string with "INFO - HELLO WORLD!"

Output with original command and logs=True:
![image](https://user-images.githubusercontent.com/12816025/84119853-ab300780-aa3d-11ea-801d-1155b02732e0.png)
^ ok - notice string with "INFO - HELLO WORLD!"

**Comment by nullhack** [comment]
This can be related to a recently merged [PR](https://github.com/apache/airflow/pull/9173).

I tried to run on the docker image `apache/airflow:master-ci` and the output was as expected.

![image](https://user-images.githubusercontent.com/11466701/84144441-d55be680-aa8a-11ea-8021-e3aee6bb6bdf.png)

If this is the case, It's related to the `auto_remove=True`, (the container gets removed before The task can get the `stdout`). If not, can be related to the `issue 1` of https://github.com/apache/airflow/issues/9164

@the-horhe If you use `DockerOperator` a lot, can you give a feedback on thi [issue/PR](https://github.com/apache/airflow/issues/9164) I proposed some days ago ? Thanks.

**Comment by the-horhe** [comment]
Tested with `auto_remove=False` - it seems it has no effect. Did't try with `apache/airflow:master-ci` though yet.
Also modified command to `/bin/bash -c \'echo "1" && sleep 5 && echo "2"\` and see only `2` in logs, so to me it seems the problem is a late container connection.

@nullhack it is the first time we try to use DockerOperator, so don't have much to say 8(

**Comment by alexkruc** [comment]
I tried to look at this issue, but I can't seem to reproduce it.
I tried running the (almost) the same code as was presented in the example:
```python
  print_hello_world = DockerOperator(
      task_id='print_hello_world',
      image='centos:latest',
      api_version='auto',
      auto_remove=True,
      command='/bin/bash -c \'echo "HELLO WORLD!"\'',
      network_mode='bridge',
      dag=dag,
  )
``` 
But I got the output of the container proper...
![image](https://user-images.githubusercontent.com/36231027/176995189-e22a018e-5460-4d53-a8bf-1edc2d72a76e.png)
Also tried running other tasks, with different containers and commands, and it all worked as expected - the output printed out immediately.

I see that the reported version of this issue was 1.10.10, there were A LOT of changes since then. 
Can someone reproduce this issue on the most recent versions and share information?
